optimum/habana/transformers/generation/utils.py

```diff
@@ -1239,8 +1243,426 @@ def contrastive_search(
         >>> tokenizer.batch_decode(outputs, skip_special_tokens=True)
         ['DeepMind Company is a company that focuses on the development and commercialization of artificial intelligence (AI). DeepMind’s mission is to help people understand and solve problems that are difficult to solve in the world today.\n\nIn this post, we talk about the benefits of deep learning in business and how it']
         ```"""
+        # init values
+        logits_processor = logits_processor if logits_processor is not None else LogitsProcessorList()
+        logits_warper = logits_warper if logits_warper is not None else LogitsProcessorList()
+        stopping_criteria = stopping_criteria if stopping_criteria is not None else StoppingCriteriaList()
+        pad_token_id = pad_token_id if pad_token_id is not None else self.generation_config.pad_token_id
```

Comment:
i don't see https://github.com/huggingface/transformers/blob/v4.40.2/src/transformers/generation/utils.py#L1960 here

Reply:
Yes, I'm using 4.38.2. Would you like me to upgrade the implementation to use 4.40.2?